### PR TITLE
[14.0][MIG] stock_picking_partner_note: Migration from 16 to 14.0

### PR DESCRIPTION
**Pull request by JuMiSanAr**
Backport from 16.0 to 14.0

### CODE REVIEW

File: stock_picking_partner_note/tests/test_stock_picking_partner_note.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import Form, common
+
+from odoo.addons.base.tests.common import DISABLED_MAIL_CONTEXT
+
+
+class StockPickingPartnerNote(common.TransactionCase):
```
Comment (rousseldenis) [inline stock_picking_partner_note/tests/test_stock_picking_partner_note.py]: @JuMiSanAr Don't forget to change this to SavepointCase

Comment (JuMiSanAr) [reply stock_picking_partner_note/tests/test_stock_picking_partner_note.py]: Done, thanks!

I've also added the missing README fragments as per pre-commit, let me know if that's ok or if they're not needed.

**Review by jbaudoux — APPROVED** [review]

File: stock_picking_partner_note/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    note = fields.Html(compute="_compute_note", store=True)
+
+    @api.depends("partner_id")
```
Comment (simahawk) [inline stock_picking_partner_note/models/stock_picking.py:18]:
```suggestion
    @api.depends("partner_id", "picking_type_id.partner_note_type_ids", "partner_id.stock_picking_note_ids")
```

File: stock_picking_partner_note/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    def _compute_note(self):
+        for picking in self:
+            picking_type_note_type_ids = picking.picking_type_id.partner_note_type_ids
+            picking_notes = picking.partner_id.stock_picking_note_ids.filtered(
+                lambda n: n.note_type_id in picking_type_note_type_ids
```
Comment (simahawk) [inline stock_picking_partner_note/models/stock_picking.py]:
```suggestion
                lambda n: n.active and n.note_type_id in picking_type_note_type_ids
```
the result must be the same even if the active_test flag is set

**Review by simahawk — APPROVED** [review]
pre-approving

File: stock_picking_partner_note/models/stock_picking.py (hunk 2 above)
Comment (JuMiSanAr) [reply stock_picking_partner_note/models/stock_picking.py:18]: When I add these compute dependencies, I get this error:

```
2024-03-04 07:17:34,994 83 WARNING odoodb odoo.modules.loading: Transient module states were reset 
2024-03-04 07:17:34,995 83 ERROR odoodb odoo.modules.registry: Failed to load registry 
Traceback (most recent call last):
  File "/odoo/src/odoo/api.py", line 794, in get
    return field_cache[record._ids[0]]
KeyError: 2663522

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/odoo/src/odoo/fields.py", line 993, in __get__
    value = env.cache.get(record, self)
  File "/odoo/src/odoo/api.py", line 797, in get
    raise CacheMiss(record, field)
odoo.exceptions.CacheMiss: 'stock.picking(2663522,).picking_type_id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/odoo/src/odoo/modules/registry.py", line 89, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/odoo/src/odoo/modules/loading.py", line 457, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check)
  File "/odoo/src/odoo/modules/loading.py", line 349, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/odoo/src/odoo/modules/loading.py", line 249, in load_module_graph
    cr.commit()
  File "<decorator-gen-7>", line 2, in commit
  File "/odoo/src/odoo/sql_db.py", line 101, in check
    return f(self, *args, **kwargs)
  File "/odoo/src/odoo/sql_db.py", line 451, in commit
    flush_env(self)
  File "/odoo/src/odoo/sql_db.py", line 78, in flush_env
    env_to_flush['base'].flush()
  File "/odoo/src/odoo/models.py", line 5472, in flush
    self.recompute()
  File "/odoo/src/odoo/models.py", line 5931, in recompute
    process(field)
  File "/odoo/src/odoo/models.py", line 5915, in process
    field.recompute(recs)
  File "/odoo/src/odoo/fields.py", line 1176, in recompute
    self.compute_value(recs)
  File "/odoo/src/odoo/fields.py", line 1198, in compute_value
    records._compute_field_value(self)
  File "/odoo/src/addons/mail/models/mail_thread.py", line 410, in _compute_field_value
    return super()._compute_field_value(field)
  File "/odoo/src/odoo/models.py", line 4079, in _compute_field_value
    odoo.fields.determine(field.compute, self)
  File "/odoo/src/odoo/fields.py", line 82, in determine
    return needle(*args)
  File "/odoo/external-src/stock-logistics-workflow/stock_picking_partner_note/models/stock_picking.py", line 21, in _compute_note
    picking_type_note_type_ids = picking.picking_type_id.partner_note_type_ids
  File "/odoo/src/odoo/fields.py", line 2495, in __get__
    return super().__get__(records, owner)
  File "/odoo/src/odoo/fields.py", line 1019, in __get__
    recs._fetch_field(self)
  File "/odoo/src/odoo/models.py", line 3075, in _fetch_field
    self._read(fnames)
  File "/odoo/src/odoo/models.py", line 3172, in _read
    self.env.cache.update(fetched, field, values)
  File "/odoo/src/odoo/api.py", line 812, in update
    field_cache.update(zip(records._ids, values))
MemoryError
2024-03-04 07:17:34,997 83 CRITICAL odoodb odoo.service.server: Failed to initialize database `odoodb`. 
Traceback (most recent call last):
  File "/odoo/src/odoo/api.py", line 794, in get
    return field_cache[record._ids[0]]
KeyError: 2663522
```

This doesn't happen if I just have `partner_id` as before, why does this happen now?
I've tried to add a default like:

```
if not picking.partner_id or not picking.picking_type_id:
    picking.note = ""
    continue
```

But I get the same error every time...

Comment (simahawk) [reply stock_picking_partner_note/models/stock_picking.py:18]: weird. Add a TODO maybe

**Comment by simahawk** [comment]
the fix for the active field should be split in its own commit and fwd ported

**Comment by simahawk** [comment]
/ocabot merge nobump

FTR small improvement here https://github.com/OCA/stock-logistics-workflow/pull/1541 that can be backported.

**Pull request merged**